pkg/sql/schemachanger/scplan/internal/rules/dep_add_index.go

```diff
@@ -71,21 +71,36 @@ func init() {
 func init() {
 
 	registerDepRule(
-		"index named right before index becomes public",
+		"primary index named right before index becomes public",
 		scgraph.SameStagePrecedence,
 		"index-name", "index",
 		func(from, to nodeVars) rel.Clauses {
 			return rel.Clauses{
 				from.Type((*scpb.IndexName)(nil)),
 				to.Type(
 					(*scpb.PrimaryIndex)(nil),
-					(*scpb.SecondaryIndex)(nil),
 				),
 				joinOnIndexID(from, to, "table-id", "index-id"),
 				statusesToPublicOrTransient(from, scpb.Status_PUBLIC, to, scpb.Status_PUBLIC),
 			}
 		},
 	)
+
+	registerDepRule(
+		"secondary index named before validation",
+		scgraph.SameStagePrecedence,
+		"index-name", "index",
+		func(from, to nodeVars) rel.Clauses {
+			return rel.Clauses{
+				from.Type((*scpb.IndexName)(nil)),
+				to.Type(
+					(*scpb.SecondaryIndex)(nil),
+				),
+				joinOnIndexID(from, to, "table-id", "index-id"),
+				statusesToPublicOrTransient(from, scpb.Status_PUBLIC, to, scpb.Status_WRITE_ONLY),
```

Comment:
1. What is the overall rational that prompts you to make this dep-rule change?
2. Specifically for this rule, why SameStagePrecedence? If your answer is "It previously was SameStagePrecedence, and I just carried it over", then I am starting to think we can change both to "Precedence"? What do you think?

Reply:
1. When we validate an index any validation error will contain the name of the index, so we need to set the index name before validation occurs. 
2. We can safely make this precedence since validation ops will have their own independent stage no matter what, so I'm okay with that change.